RecoLocalCalo/HcalRecAlgos/interface/HcalSimpleRecAlgo.h

```diff
@@ -111,8 +108,13 @@ class HcalSimpleRecAlgo {
 
   std::auto_ptr<PulseShapeFitOOTPileupCorrection> psFitOOTpuCorr_;
   
+  std::auto_ptr<PedestalSub> pedSubFxn_;
+
   // S.Brandt Feb19 : Add a pointer to the HLT algo
   std::auto_ptr<HcalDeterministicFit> hltOOTpuCorr_;
+
+  // Speed up the code a little bit by not allocating a vector every time
+  mutable std::vector<double> vectorBuffer_;
```

Comment:
there are several more allocations in the methods called internally.
So, if a fix needs to be done, better do it properly.

Your change with a mutable opens a possibility to pass information between hits.
This is not good.

Cleaning and clearing is apparently done correctly now, but this makes the code much more fragile in the future developments.